sdk/servicebus/service-bus/test/invalidParameters.spec.ts

```diff
@@ -403,6 +428,72 @@ describe("Invalid parameters in SessionReceiver", function(): void {
     should.equal(caughtError && caughtError.message, `Missing parameter "fromSequenceNumber"`);
   });
 
+  it("RegisterMessageHandler: Missing onMessage in SessionReceiver", async function(): Promise<
+    void
+  > {
+    let caughtError: Error | undefined;
+    try {
+      await sessionReceiver.registerMessageHandler(undefined as any, undefined as any);
+    } catch (error) {
+      caughtError = error;
+    }
+    should.equal(caughtError && caughtError.name, "TypeError");
+    should.equal(caughtError && caughtError.message, `Missing parameter "onMessage"`);
+  });
+
+  it("RegisterMessageHandler: Wrong type for onMessage in SessionReceiver", async function(): Promise<
+    void
+  > {
+    let caughtError: Error | undefined;
+    try {
+      await sessionReceiver.registerMessageHandler("somestring" as any, "somethingelse" as any);
+    } catch (error) {
+      caughtError = error;
+    }
+    should.equal(caughtError && caughtError.name, "TypeError");
+    should.equal(
+      caughtError && caughtError.message,
+      `The parameter 'onMessage' must be of type 'function'.`
+    );
+  });
+
+  it("RegisterMessageHandler: Missing onError in SessionReceiver", async function(): Promise<void> {
+    let caughtError: Error | undefined;
+    try {
+      await sessionReceiver.registerMessageHandler(
+        async () => {
+          /** */
+        },
+        undefined as any
+      );
+    } catch (error) {
+      caughtError = error;
+    }
+    should.equal(caughtError && caughtError.name, "TypeError");
```

Comment:
These should statements seem confusing to me, does `caughtError && caughtError.name` resolve to caughtError.name?

Reply:
Yes, try it out in a test file.

If `caughtError` is not falsy (null, undefined, false, 0), then that will resolve to `caughtError.name` else whatever is the value for `caughtError`